post_office/mail.py

```diff
@@ -270,7 +272,26 @@ def send(email):
     Email.objects.filter(id__in=email_ids).update(status=STATUS.sent)
 
     email_ids = [email.id for (email, e) in failed_emails]
-    Email.objects.filter(id__in=email_ids).update(status=STATUS.failed)
+
+    numbers_mails_get_back_in_queue = 0
+    # Retry is enabled
+    if get_max_retry() > 0:
+        emails_failed = Email.objects.filter(id__in=email_ids)
+        for email in emails_failed:
+            if email.number_of_retries < get_max_retry():
+                email.status = STATUS.requeued
+                email.scheduled_time = now() + get_time_delta_to_retry()
+                if email.number_of_retries is None:
+                    email.number_of_retries = 1
+                else:
+                    email.number_of_retries = email.number_of_retries + 1
+                numbers_mails_get_back_in_queue += 1
+            else:
+                email.status = STATUS.failed
+
+            email.save()
```

Comment:
We should use `bulk_update()` for this. https://docs.djangoproject.com/en/3.0/ref/models/querysets/#bulk-update

Reply:
ok
Same thing for requeued email